File: math/russian_peasant/java/RussianPeasant.java

public class RussianPeasant {
	
	public static int calculate(int a, int b) {
		int x = a;
	    int y = b;
	    int results = 0;
	    
	    while (x > 0) {
	    	if (x % 2 == 1) results = results + y;
	    	y = y << 1;
	    	x = x >> 1;
	    }

		return results;
	}
	
	public static void main(String[] args) {
		int x = RussianPeasant.calculate(3, 4);
		System.out.println(x);
	}
}

File: math/russian_peasant/php/russian_peasant.php
<?php

function russianPeasant($a, $b)
{
    $result = 0;
    while ($a > 0) {
        if ($a % 2 == 1) {
            $result += $b;
        }
        $a >>= 1;
        $b <<= 1;
    }
    
    return $result;
}

if (russianPeasant(5, 15) != 75) {
    echo "Test failed.";
} else {
    echo "Test passed.";
}
